components/coding/ProfileHeaderComponent.tsx

```diff
@@ -0,0 +1,125 @@
+import { useQuery } from "@apollo/client";
+import { format } from "date-fns";
+import React, { useState } from "react";
+
+import { FETCH_TOTAL_USER_BADGES_COUNT } from "../../graphql/fetchTotalUserBadgesCount";
+import {
+  FetchUserBadgesCountResponse,
+  FETCH_USER_BADGES_COUNT,
+} from "../../graphql/fetchUserBadgesCount";
+
+import {
+  FetchUserProfileDataResponse,
+  FETCH_USER_PROFILE_DATA,
+  UserProfileData,
+} from "../../graphql/fetchUserProfile";
+
+export type UserProfileSectionProps = {
+  user: any;
```

Comment:
let's try to get in the habit of not using the any type unless we have to... I know we're doing it across the codebase but lets try defining a type for this user object so we're using Typescript properly

Reply:
created a User type. great idea.